File: app/src/main/java/com/noblemajesty/marvel/models/getCharacters/Comics.java
package com.noblemajesty.marvel.models.getCharacters;

import java.util.List;
import android.os.Parcel;
import android.os.Parcelable;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Comics implements Parcelable
{

  @SerializedName("available")
  @Expose
  private Integer available;
  @SerializedName("collectionURI")
  @Expose
  private String collectionURI;
  @SerializedName("comicItems")
  @Expose
  private List<ComicItem> comicItems = null;
  public final static Parcelable.Creator<Comics> CREATOR = new Creator<Comics>() {


    @SuppressWarnings({
        "unchecked"
    })
    public Comics createFromParcel(Parcel in) {
      return new Comics(in);
    }

    public Comics[] newArray(int size) {
      return (new Comics[size]);
    }

  };

  protected Comics(Parcel in) {
    this.available = ((Integer) in.readValue((Integer.class.getClassLoader())));
    this.collectionURI = ((String) in.readValue((String.class.getClassLoader())));
    in.readList(this.comicItems, (ComicItem.class.getClassLoader()));
  }

  public Integer getAvailable() {
    return available;
  }

  public String getCollectionURI() {
    return collectionURI;
  }

  public List<ComicItem> getComicItems() {
    return comicItems;
  }

  public void writeToParcel(Parcel dest, int flags) {
    dest.writeValue(available);
    dest.writeValue(collectionURI);
    dest.writeList(comicItems);
  }

  public int describeContents() {
    return 0;
  }

}

File: app/src/main/java/com/noblemajesty/marvel/models/getCharacters/Comics.kt
package com.noblemajesty.marvel.models.getCharacters

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Comics protected constructor(`in`: Parcel) : Parcelable {

    @SerializedName("available")
    @Expose
    val available: Int?
    @SerializedName("collectionURI")
    @Expose
    val collectionURI: String
    @SerializedName("comicItems")
    @Expose
    val comicItems: List<ComicItem>? = null

    init {
        this.available = `in`.readValue(Int::class.java.classLoader) as Int
        this.collectionURI = `in`.readValue(String::class.java.classLoader) as String
        `in`.readList(this.comicItems, ComicItem::class.java.classLoader)
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(available)
        dest.writeValue(collectionURI)
        dest.writeList(comicItems)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {
        val CREATOR: Parcelable.Creator<Comics> = object : Parcelable.Creator<Comics> {


            override fun createFromParcel(`in`: Parcel): Comics {
                return Comics(`in`)
            }

            override fun newArray(size: Int): Array<Comics?> {
                return arrayOfNulls(size)
            }

        }
    }

}

